auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -589,18 +715,29 @@ public boolean verifyEvidence(final Evidence evidence) throws RealmUnavailableEx
             Assert.checkNotNullParam("evidence", evidence);
 
             if (ElytronMessages.log.isTraceEnabled()) {
-                final LoadedIdentity loadedIdentity = loadIdentity(false, true);
-                ElytronMessages.log.tracef("Trying to authenticate identity %s using FileSystemSecurityRealm",
-                        (loadedIdentity != null) ? loadedIdentity.getName() : "null");
+                try {
+                    final LoadedIdentity loadedIdentity = loadIdentity(false, true);
+                    ElytronMessages.log.tracef("Trying to authenticate identity %s using FileSystemSecurityRealm", (loadedIdentity != null) ? loadedIdentity.getName() : "null");
+                } catch (RealmUnavailableException e) {
+                    if (e.getCause() instanceof RealmUnavailableException) {
```

Comment:
Can there be a different cause than RealmUnavailableException here and below?

Reply:
I think that if the error is not related to the invalid signature, it just runs `throw new RealmUnavailableException()` but if it's related to the signature, it runs `throw new RealmUnavailableException(ElytronMessages.log.invalidIdentitySignature(name))` so that the cause is also RealmUnavailableException.

I think for most cases the cause should be null

Reply:
Wherever the signature is determined to be invalid, instead of just throwing a `RealmUnvailableException` there, you can wrap the underlying exception with a `RealmUnavailableException` so then the cause itself will be something else. See `RealmUnavailableException(final Throwable cause)`.